### DISCUSSION TITLE
Forwarding remote tracks received from one peer to all other peers .

### DISCUSSION THREAD

**Discussion by manoj-thamizhan** [aiortc/aiortc#823] [Q&A] [answered]
Hi @jlaine  ,
 First of all thank you for this amazing frame work

I'm facing issue with forwarding tracks received from one peer to all other peers which are running in separate threads

1. I'm receiving remote tracks and i'm adding to global set from one peer connection
2. When new peer joins the i'm adding all the tracks in the peer with addtrack
3. The tracks are received in remote side but not playing and not receiving any frames except for only one user i,e the firt user to receive the remote track and playing but other peer connections not receiving any frames and the track is muted not unmutes

thanks a lot in advance

**Comment by LeoNorval** [comment, marked as answer]
Maybe I can help:

It seems as though you are trying to add the same track to multiple "consumers", i.e. PeerConnection, MediaRecorder/MediaBlackhole or other sinks. 
However, each track will release the frame it has as soon as it's recv method is called, so adding the same track instance to multiple consumers will result in most of them waiting endlessly for a frame.
To be able to relay the track to multiple consumers, use the MediaRelay class to receive a relayed track for each consumer.
An example of it's usage can be found here:
[examples/webcam/webcam.py](https://github.com/aiortc/aiortc/blob/f4e3049875142a18fe32ad5f2c052b84a3112e30/examples/webcam/webcam.py#L22)

**Reply by manoj-thamizhan** [reply]
Thank You @LeoNorval  it worked as you've mentioned 

Thank You once again 😊
